docs/Configuration.md

```diff
@@ -31,6 +31,25 @@ module.exports = async () => {
 };
 ```
 
+Or through TypeScript:
+
+```ts
+// jest.config.ts
+import type { Config } from '@jest/types';
+
+//Sync object
+export = {
+  verbose: true,
+} as Config.InitialOptions;
```

Comment:
```suggestion
const config: Config.InitialOptions = {
  verbose: true
};

export = config;
```

Casting is subtly different from annotating: it'll have the compiler warn for excess properties but not for missing required properties, including with nested objects.

While I think most of our options (if not all) are optional, I think it's better to encourage using an annotation (see #9570 & co for examples)- keep in mind that you can use module augmentation to include properties from other packages like `ts-jest` which might have required properties.

Reply:
@G-Rath Make sense, thank you for input. I have some tests to fix and I will push a commit with this changes.